Source/Ziko/Private/Characters/Player/WizardCharacter.cpp

```diff
@@ -36,14 +34,50 @@ void AWizardCharacter::SetupPlayerInputComponent(UInputComponent* PlayerInputCom
 void AWizardCharacter::BaseAttack()
 {
 	Super::BaseAttack();
+	if (AttackType != EAttackType::AT_None) return;
+	
+	ABaseWeapon* const MagicWand = GetPrimaryWeapon();
+	if (!MagicWand) return;
+
+	const int8 AttackEnergyCost = MagicWand->GetAttackCost(EAttackType::AT_Basic);
+	if (EnergyVal < AttackEnergyCost) return;
+	
+	AttackType = EAttackType::AT_Basic;
+	GetWorldTimerManager().SetTimer(AttackTimer, MagicWand, &ABaseWeapon::BaseAttack, BaseAttackWaitRate, false);
+	
+	EnergyVal -= AttackEnergyCost;
```

Comment:
I dont think we need this , since we can use montages and all characters can use the same base implementation - play an animation montage and take some energy away which is retrieved from the weapon